modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/objectDefinition.ts

```diff
@@ -17,5 +17,5 @@ export function normalizeName(str: string) {
 }
 
 export function findObjectDefinitionById(objectDefinitions: ObjectDefinition[], id: number) {
-	return 	objectDefinitions.find(objectDefinition => objectDefinition.id === id);
+	return 	objectDefinitions.find(objectDefinition => objectDefinition.id === Number(id));
```

Comment:
can you explain to me why this was necessary? because from the type the id can only be a number on this case right?

Reply:
I did this because even though the id is defined as a number, it is being set as a string when the user selects an object.

![Screenshot from 2025-02-25 10-40-03](https://github.com/user-attachments/assets/dd962e02-588f-4001-abe9-e074a392865d)

![Screenshot from 2025-02-25 10-42-30](https://github.com/user-attachments/assets/9948dd41-127a-446b-b3cc-d70cb1f75cf0)

Reply:
oh i see, but if that's the case, the id should be transformed to number before passing to the `findObjectDefinitionById`, don't you think? because how it is right now the function is expecting to receive a number but gets a string, and only inside the function the change of type happens

Reply:
also how this was working before if the value comes as a string? or is only coming as a string now after the story? i'm trying to understand the context here to share the best suggestion

Reply:
did you have any insights about this @cgmonte during the refinement? since on the stories we are working now the id is always a string and we also send a string, i'm finding this behavior a little bit odd; but i guess is just the way the backend is sending things on this case, and its necessary to convert to a number to make the comparison

Reply:
Yeah, it makes sense to cast the `id` before passing it to the function. As for how it was working before, the value was coming in as a string, but in all of the interfaces, it is defined as a number.

Reply:
hi @DavysonMelo! @cgmonte explained to me why you both decided to do this way and i understand now! you can leave as it is, thanks for your patience on this!